ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/composables/Footer.kt

```diff
@@ -154,3 +171,52 @@ private fun FooterPreview() {
         viewModel = MockViewModel(TestData.template2Offering)
     )
 }
+
+@Composable
+fun RowScope.AdaptiveView(views: List<@Composable () -> Unit>) {
+    var selectedIndex by remember { mutableStateOf(0) }
+    var maxSize by remember { mutableStateOf(0) }
+    var viewSizes by remember { mutableStateOf(List(views.size) { 0 }) }
+
+    Box(
+        modifier = Modifier
+            .fillMaxWidth()
+            .onGloballyPositioned { coordinates ->
+                maxSize = coordinates.size.width
+            }
+            .align(Alignment.CenterVertically)
+            .background(Color.Green.copy(alpha = 0.5f))
+    ) {
+        // Offscreen measurement
+        Box(
+            Modifier
+                .graphicsLayer(alpha = 0.0f)
+                .fillMaxWidth()
+                .align(Alignment.Center)
+        ) {
+            views.forEachIndexed { index, view ->
+                Box(
+                    Modifier.layout { measurable, _ ->
+                        val placeable = measurable.measure(Constraints())
```

Comment:
Huh interesting... I have never actually used many of these. But reading a bit about it, it makes sense.